src/csync/csync_private.h

```diff
@@ -99,8 +101,17 @@ struct OCSYNC_EXPORT csync_s {
       void *checksum_userdata = nullptr;
 
   } callbacks;
-  c_strlist_t *excludes = nullptr;
-  
+
+  c_strlist_t *excludes = nullptr; /* list of individual patterns collected from all exclude files */
+  struct {
+      QRegularExpression regexp_exclude;
+      QRegularExpression regexp_exclude_and_remove;
+      /* FIXME ^^ could we use one regexp by using captures? (exclude|these)|(exclude|and|remove|these)
+       * and then check which capture is nonempty? Unclear whether that'd be faster*/
+      c_strlist_t *list_patterns_with_slashes = nullptr;
```

Comment:
I guess this should be a QVector or something

Reply:
@ogoffart I copy pasted code from the previous implementation which is using c_strlist_t. I agreed with @ckamm that we keep the code un-cleaned up for now and try to get this PR in. Cleaning can be done in a follow up (assuming tests etc still pass)